app/models/factory.rb

```diff
@@ -1,76 +1,44 @@
-      end
-      assos.each do |asso|
-        Asso.create_new_asso_and_relation(asso, new_factory)
-      end
-    else
-      same_factory = same_factory(name, traits)
-      return same_factory_overwrite(assos, Factory.find(same_factory["id"])) if same_factory
-      new_factory = Factory.create(name: name)
-      traits.each do |trait|
-        if Factory.same_trait_exist?(trait, name)
-          existing_trait = Trait.where(name: trait).select { |tr| tr.factories.first.name == name }.first
-          TraitRelation.create_new_trait_relation(new_factory, existing_trait)
-        else
-          Trait.create_new_trait_and_relation(new_factory, trait)
-        end
-        assos.each do |asso|
-          Asso.create_new_asso_and_relation(asso, new_factory)
-        end
-      end
-    end
-    REDIS.sadd("factory_with_traits", { factory_name: new_factory.name, traits: new_factory.traits.map(&:name).to_s, id: new_factory.id }.to_json)
-    new_factory
-  end
 
-  # Return same name and same traits having factory.
-  def self.same_factory(name, trait_names)
-    trait_names = trait_names.to_s
-    result = REDIS.smembers("factory_with_traits").find do |fwt|
-      factory_hash = JSON.parse(fwt)
-      factory_hash["factory_name"] == name && factory_hash["traits"] == trait_names
-    end
-    JSON.parse(result) if result
-  end
+    if is_first_look_factory?
+      new_factory = Factory.create(name: factory_name)
+      Trait.create_new_traits_and_relations(new_factory, traits)
+      Asso.create_new_assos_and_relations(new_factory, assos)
+    else
+      same_factory = search_same_factory(factory_name, traits)
+      # We can get factory's assos only when it executed as "subject" (not as "asso" of other factory!).
+      # So, We must overwrite it's asso and asso_relation in case the factory registered when it executed as other factory's asso before.
+      return overwrite_same_factory_asso(assos, Factory.find(same_factory["id"])) unless same_factory.nil?
 
-  # Return same factory. If if it is no asoociarion, overwrite new assos.
-  def self.same_factory_overwrite(assos, same_factory)
-    if assos == same_factory.assos
-      return same_factory
-    elsif !assos.empty? && same_factory.assos.empty?
-      assos.each do |asso|
-        Asso.create_new_asso_and_relation(asso, same_factory)
-      end
+      new_factory = Factory.create(name: factory_name)
+      Trait.add_new_traits_and_relations(new_factory, traits)
+      Asso.create_new_assos_and_relations(new_factory, assos)
     end
-    same_factory.reload
-  end
-
-  # Return whether same name and parent factory trait exist.
-  def self.same_trait_exist?(trait, factory_name)
-    !REDIS.sadd("traits", { name: trait, factory_name: factory_name }.to_json)
+    # Register concat factory and traits for uniqueness inspection @#search_same_factory
+    REDIS.sadd("factory_with_traits", {
+                                        factory_name: new_factory.name,
```

Comment:
Style/IndentHash: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.